doc/man1/s_client.pod

```diff
@@ -409,6 +412,11 @@ given as a hexadecimal number without leading 0x, for example -psk
 1a2b3c4d.
 This option must be provided in order to use a PSK cipher.
 
+=item B<-psk_session file>
+
+Use the pem encoded SSL_SESSION data stored in B<file> as the basis of a PSK.
+Note that this will only work if TLSv1.3 is negotiated.
```

Comment:
Is that one that's generated by using -sess_out?

What is the difference between using -psk_session and -sess_in?

Reply:
> Is that one that's generated by using -sess_out?

It could be generated by "-sess_out", or from the output from `PEM_write_bio_SSL_SESSION()` (or similar). The SSL_SESSION itself could be generated from a previous handshake or constructed "manually" via `SSL_SESSION_new()` and then setting all of the various parameters (master key etc).

> What is the difference between using -psk_session and -sess_in?

The identity used is different. With `-sess_in` the identity is the opaque ticket data held inside the session itself. With `-psk_session` the identity is set via `-psk_identity`.

Reply:
> So -psk_identity is required when using this option? I think I can at least get the same behaviour where both set it to the same value or where both don't set it.

That option has a default value of "Client_identity" so not setting on both sides should still work.

> I'm not sure how to tell the difference in the output of s_server / s_client if PSK is used or not. At least s_client says the psk identity is and psk identity hint are None in all cases, but if the identity is not the same I get an error.

If successful it will appear like a session resumption. The hint is never used in TLSv1.3. s_client should report the identity - sounds like a possible bug.